src/jobs/prepare_complaint_raw_dataset/__init__.py

```diff
@@ -123,10 +116,12 @@ def analyze(spark, sc, config):
         file_paths = _generate_part_files_location(config,curr_ckpt_info)
         curr_ckpt_info.run_status = 'RUNNING'
         curr_ckpt_info.update_in_db(spark,config)
+        
         for fpath in file_paths:
```

Comment:
For loop will be a bottleneck. Rather opt for reading folder, or some other strategy where workloads can be distributed.